tad_pathways/tad.py

```diff
@@ -0,0 +1,189 @@
+
+    def _create_filenames(self):
+        """
+        Create several variables that correspond to the specific snp list
+        """
+
+        if not os.path.exists(self.base_dir):
+            os.makedirs(self.base_dir)
+
+        # File to compile RSid lookup results
+        self.snp_location_file = os.path.join(
+            self.base_dir, "{}_location.tsv".format(self.snp_list_name)
+        )
+
+        # Files for TAD based genesets - all genes that fall in candidate variant TADs
+        self.genelist_file = os.path.join(
+            self.base_dir, "{}_tad_genelist.tsv".format(self.snp_list_name)
+        )
+
+        # Files from WebGestalt output
+        self.pathway_p_values_file = os.path.join(
+            self.base_dir, "{}_pvals.tsv".format(self.snp_list_name)
+        )
+
+        # Files used to summarize results
+        self.nearest_gene_file = os.path.join(
+            self.base_dir, "{}_tad_genelist_nearest_gene.tsv".format(self.snp_list_name)
+        )
+
+        self.gene_evidence_file = os.path.join(
+            self.base_dir, "{}_gene_evidence.csv".format(self.snp_list_name)
+        )
+
+        self.evidence_summary_file = os.path.join(
+            self.base_dir, "{}_gene_evidence_summary.tsv".format(self.snp_list_name)
+        )
+
+        self.all_pathways_evidence = os.path.join(
+            self.base_dir,
+            "{}_all-sig-pathways_gene_evidence.csv".format(self.snp_list_name),
+        )
+
+        self.all_pathways_summary = os.path.join(
+            self.base_dir,
+            "{}_all-sig-pathways_gene_evidence_summary.tsv".format(self.snp_list_name),
+        )
+
+    def build_snp_list(self):
+        command_list = [
+            "Rscript",
+            "--vanilla",
+            "scripts/build_snp_list.R",
+            "--snp_file",
+            self.snp_list_location,
+            "--output_file",
+            self.snp_location_file,
+        ]
+        subprocess.call(command_list)
+
+    def build_custom_tad_genelist(self):
+        command_list = [
+            "python",
+            "scripts/build_custom_tad_genelist.py",
+            "--snp_data_file",
+            self.snp_location_file,
+            "--output_file",
+            self.genelist_file,
+        ]
+
+        if self.remove_hla:
+            command_list += ["--remove_hla_tad"]
+
+        subprocess.call(command_list)
+
+    def webgestalt_run(self):
+        command_list = [
+            "Rscript",
+            "--vanilla",
+            "scripts/webgestalt_run.R",
+            "--tad_genelist_file",
+            self.genelist_file,
+            "--output_name",
+            self.snp_list_name,
+            "--output_directory",
+            self.base_dir,
+        ]
+        subprocess.call(command_list)
+
+    def check_webgestalt(self):
+        output_pval_file = os.path.join(
+            self.base_dir, "{}_pvals.tsv".format(self.snp_list_name)
+        )
+        if os.path.exists(output_pval_file):
```

Comment:
If you want, lines 136-139 can be simplified as:
```
return os.path.exists(output_pval_file)
```

Reply:
:tada: